util/logutil/log.go

```diff
@@ -96,6 +96,32 @@ const (
 // SlowQueryLogger is used to log slow query, InitLogger will modify it according to config file.
 var SlowQueryLogger = log.L()
 
+// GeneralLogLogger is used to log general log
+var GeneralLogLogger = log.L()
+
+var generalLogMgr *GeneralLogManager
+
+// PutGeneralLogOrDrop sends the general log entry to the logging goroutine
+// The entry will be dropped once the channel is full
+func PutGeneralLogOrDrop(entry *GeneralLogEntry) {
+	select {
+	case generalLogMgr.logEntryChan <- entry:
+	default:
+		// When logEntryChan is full, the system resource is under so much pressure that the logging system capacity
+		// is reduced. We should NOT output another warning log saying that we are dropping a general log, which will
+		// add more pressure on the system load.
+		// TODO(dragonly): add prometheus counter for dropped log count?
+		generalLogMgr.droppedLogCount++
+	}
+}
+
+// InitGeneralQueryLog initialize general query logger, which will starts a format & logging worker goroutine
+// General query logs are sent to the worker through a channel, which is asynchronously flushed to logging files.
+func InitGeneralQueryLog() {
+	GeneralLogLogger := newGeneralLogLogger()
+	generalLogMgr = newGeneralLog(GeneralLogLogger)
```

Comment:
TiDB rarely use XxxManager as a naming. Maybe you should think about some other names, like:

GeneralLogger (where the zapLogger is a field inside)